src/Host/Client/Impl/Host/RemoteCredentialsDecorator.cs

```diff
@@ -14,17 +14,19 @@
 
 namespace Microsoft.R.Host.Client.Host {
     internal class RemoteCredentialsDecorator : ICredentialsDecorator {
-        private readonly IServiceContainer _services;
         private volatile Credentials _credentials;
         private readonly AsyncReaderWriterLock _lock;
         private readonly string _authority;
         private readonly string _workspaceName;
+        private readonly IMainThread _mainThread;
+        private readonly ISecurityService _security;
 
-        public RemoteCredentialsDecorator(string credentialAuthority, string workspaceName, IServiceContainer services) {
-            _services = services;
+        public RemoteCredentialsDecorator(string credentialAuthority, string workspaceName, IMainThread mainThread, ISecurityService security) {
```

Comment:
Intent to pass `IServiceContainer` here was to be able to handle that `//TODO` below when cred storage cannot be accessed (we had actual bug on it). We may need to show some UI or log stuff.